keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Multinomial.java

```diff
@@ -0,0 +1,134 @@
+package io.improbable.keanu.distributions.discrete;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.math3.special.Gamma;
+import org.nd4j.linalg.util.ArrayUtil;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import io.improbable.keanu.distributions.DiscreteDistribution;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.TensorShapeValidation;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+
+public class Multinomial implements DiscreteDistribution {
+
+    private final IntegerTensor n;
+    private final DoubleTensor p;
+    private final int numCategories;
+
+    public static Multinomial withParameters(IntegerTensor n, DoubleTensor p) {
+        return new Multinomial(n, p);
+    }
+
+    /**
+     * https://en.wikipedia.org/wiki/Multinomial_distribution
```

Comment:
We can use the JavaDoc tag `@see` for references. (e.g. `@see <a href="https://en.wikipedia.org/wiki/Multinomial_distribution">Multinomial Distribution</a>` or if you don't `a` tag, just put the link in quotes)